src/base64Utils.js

```diff
@@ -4,6 +4,11 @@
  * SPDX-License-Identifier: MIT
  */
 
+
+#if GZIP_EMBEDDINGS && !WASM_ASYNC_COMPILATION
+#error GZIP_EMBEDDINGS does not work with synchronous WASM compilation (because DecompressionStream is async by nature)
```

Comment:
Can you report this error instead in `link.py` alongside the `GZIP_EMBEDDINGS is not currently meaningful except in SINGLE_FILE mode` errro?